src/test/scala/mesosphere/marathon/api/akkahttp/v2/PodsControllerTest.scala

```diff
@@ -466,6 +470,101 @@ class PodsControllerTest extends UnitTest with ScalatestRouteTest with RouteBeha
         (jsonResponse \ 1).get should have(podId("another_pod"))
       }
     }
+
+    "update a simple single-container pod from docker image w/ shell command" in {
+      implicit val podSystem = mock[PodManager]
+      val f = Fixture()
+      val controller = f.controller()
+
+      val deploymentPlan = DeploymentPlan.empty
+      f.podManager.update(any, eq(false)).returns(Future.successful(deploymentPlan))
+
+      val postJson = """
+                       | { "id": "/mypod", "networks": [ { "mode": "host" } ], "containers": [
+                       |   { "name": "webapp",
+                       |     "resources": { "cpus": 0.03, "mem": 64 },
+                       |     "image": { "kind": "DOCKER", "id": "busybox" },
+                       |     "exec": { "command": { "shell": "sleep 1" } } } ] }
+                     """.stripMargin
+      val entity = HttpEntity(postJson).withContentType(ContentTypes.`application/json`)
+      val request = Put("/mypod")
+        .withEntity(entity)
+        .withHeaders(`Remote-Address`(RemoteAddress(InetAddress.getByName("192.168.3.12"))))
+
+      request ~> controller.route ~> check {
+        response.status should be(StatusCodes.OK)
+        response.header[Headers.`Marathon-Deployment-Id`].value.value() should be(deploymentPlan.id)
+        val jsonResponse = Json.parse(responseAs[String])
+        jsonResponse should have(
+          podId("/mypod"),
+          executorResources(cpus = 0.1, mem = 32.0, disk = 10.0),
+          noDefinedNetworkname,
+          networkMode(raml.NetworkMode.Host)
+        )
+      }
+    }
+
+    "do not update the pod if pod name is wrong" in {
+      implicit val podSystem = mock[PodManager]
+      val f = Fixture()
+      val controller = f.controller()
+
+      val podId = PathId("/unknownpod")
+
+      val deploymentPlan = DeploymentPlan.empty
+      f.podManager.update(any, eq(false)).returns(Future.failed(new PodNotFoundException(podId)))
+
+      val postJson = """
+                       | { "id": "/unknownpod", "networks": [ { "mode": "host" } ], "containers": [
+                       |   { "name": "webapp",
+                       |     "resources": { "cpus": 0.03, "mem": 64 },
+                       |     "image": { "kind": "DOCKER", "id": "busybox" },
+                       |     "exec": { "command": { "shell": "sleep 1" } } } ] }
+                     """.stripMargin
+      val entity = HttpEntity(postJson).withContentType(ContentTypes.`application/json`)
+      val request = Put(podId.toString)
+        .withEntity(entity)
+        .withHeaders(`Remote-Address`(RemoteAddress(InetAddress.getByName("192.168.3.12"))))
+
+      request ~> controller.route ~> check {
+        rejection shouldBe a[EntityNotFound]
+        inside(rejection) {
+          case r: EntityNotFound => r.message.message shouldEqual s"Pod '/unknownpod' does not exist"
```

Comment:
Checkout the changes in https://github.com/mesosphere/marathon/pull/5767. One of us will have to adopt this 🙂.